### ISSUE TITLE
CSS `position: drag`

### ISSUE THREAD

**Issue by yashrajbharti** [w3c/csswg-drafts#11383]
## Why

Creating draggable elements currently requires significant JavaScript, adding complexity and potential performance issues. A native CSS property can make this feature available directly via CSS, simplifying implementation and improving performance.

## Motivation

Draggable elements are widely used in modern web applications, such as:

- **Design tools**: Figma allows users to drag and position elements freely.
- **Media players**: YouTube captions can be dragged to reposition them.
- **Interactive tools**: Google Labs features draggable UI components.

<img width="1238" alt="Youtube" src="https://github.com/user-attachments/assets/cb1384b0-d78a-43fa-9b2c-50080fddb1e1" />
<img width="1206" alt="Figma" src="https://github.com/user-attachments/assets/4bf98520-e61c-493f-a809-f87a798498cd" />
<img width="1682" alt="Google Labs" src="https://github.com/user-attachments/assets/7fb8aee1-799b-49a2-96a5-7fff9a6a8738" />

Currently, implementing such functionality requires JavaScript, which increases complexity, reduces performance, and can lead to inconsistencies in behavior. A native CSS solution would eliminate these issues.

### Syntax

```css
/* Enable draggable behavior */
.element {
  position: drag;
}
```

### Behavior

1. **Draggable Activation**: `position: drag` makes an element draggable.
2. **Dynamic Position Updates**: Updates `top` and `left` values as the element moves.
3. **Optional Enhancements**: Future extensions like `drag-axis` or `drag-snap` could provide additional control.

### Hack Example: Interactive Ball ([CodePen](https://codepen.io/driftblaze/pen/emOgZaj))

Current draggable behavior is achieved using `resize: both` as a workaround, which can also be replicated with JavaScript. With `position: drag`, similar functionality could be achieved natively in CSS.

## Related Work

I have been following the work on [CSS Cursor Positioning](https://github.com/w3c/csswg-drafts/issues/10574) and can say it is quite close to my proposal. It complements this idea well: while their proposal introduces `cursor-x` and `cursor-y` values for animation based on cursor movement, mine focuses on user-driven dragging of elements. Together, they could offer a comprehensive set of tools for interactive designs in CSS.

**Comment by Loirooriol** [comment]
This doesn't seem like a new positioning mode. Instead, you want the browser to set `top` and `left` (or `inset-block-start` and `inset-block-end`?). But ultimately an author could want this to be based either on relative, absolute, or fixed positioning. So it can't just be a new lone `position` value, I would add a new property instead.

**Comment by yashrajbharti** [comment]
You're right! The more I think about it, having a separate drag property instead of a new positioning mode makes a lot of sense. 
Something in the lines of:
```css
/* Enable draggable behavior */
.element {
  position: absolute; /* or relative, fixed, etc. */
  drag: enabled;      /* New property to allow dragging */
}
```
